airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/error_handlers/http_response_filter.py

```diff
@@ -42,8 +42,10 @@ def __post_init__(self, parameters: Mapping[str, Any]) -> None:
         if isinstance(self.action, str):
             self.action = ResponseAction[self.action]
         self.http_codes = self.http_codes or set()
-        if isinstance(self.predicate, str):
+        if self.predicate and isinstance(self.predicate, str):
             self.predicate = InterpolatedBoolean(condition=self.predicate, parameters=parameters)
+        else:
+            self.predicate = ""
```

Comment:
I find this surprising. shouldn't we keep the `predicate` passed as parameter if it's an `InterpolatedBoolean`?

Reply:
ref